src/Components/Button.jsx

```diff
@@ -0,0 +1,19 @@
+import React from 'react';
+import { generateListToken, getListToken } from '../lib/tokens';
+import { useHistory } from 'react-router-dom';
+
+export default function Button({ token, setToken }) {
+  let history = useHistory();
+
+  const handleClick = () => {
+    generateListToken();
+    setToken(getListToken());
```

Comment:
Since `generateListToken` accesses `localStorage` and sets `token` there, you could consider calling `generateListToken` from inside `setToken` and make one less call to `localStorage`. Again, not a blocker!